### PR TITLE
Fix Posts cutoff

### PR DESCRIPTION
**Pull request by enahum**
#### Summary
This PR fixes the text of the posts being cut off in some cases after replying to a post or even after sending a new message in the channel's post list, the main reason was the `PixelRatio` changing the dimension of the emoji's causing a disruption in the post layout.

Other fixes in this PR:
* Post was being re-rendered without need cause of the `config` and `license` props, those props were used to mark the post as it cannot be edited anymore (but that function was actually broken) now we handle the time a post can be edited in a different more effective way.
* Added an `overflow: hidden` to the emoji picker so the `FlatList` and `SectionList` can recycle the Views
* No system emoji's were being added and unnecessary request made to the server that returned an error
* WhyDidYouUpdate util wasn't working so now it uses the element as an argument

### CODE REVIEW

File: app/components/post/index.js [hunk 1]
```diff
@@ -78,33 +77,36 @@ function makeMapStateToProps() {
...
         let canDelete = false;
         let canEdit = false;
+        let canEditUntil = -1;
         if (post) {
             canDelete = canDeletePost(state, config, license, currentTeamId, currentChannelId, currentUserId, post, isAdmin, isSystemAdmin);
             canEdit = canEditPost(state, config, license, currentTeamId, currentChannelId, currentUserId, post);
+            if (canEdit && license.IsLicensed === 'true' &&
+                (config.AllowEditPost === General.ALLOW_EDIT_POST_TIME_LIMIT || (config.PostEditTimeLimit !== -1 && config.PostEditTimeLimit !== '-1'))
+            ) {
+                canEditUntil = post.create_at + (config.PostEditTimeLimit * 1000) + 1000;
```
Comment (saturninoabril) [inline app/components/post/index.js]: Why additional `+ 1000`?

Comment (enahum) [reply app/components/post/index.js]: just following what the original function does https://github.com/mattermost/mattermost-redux/blob/master/src/utils/post_utils.js#L86

Comment (saturninoabril) [reply app/components/post/index.js]: I can't exactly tell if that's same.  Changes above seems to add 1 sec on top of the `PostEditTimeLimit` set value.  I'd like to think that it should only be:
```
canEditUntil = post.create_at + (config.PostEditTimeLimit * 1000);
```

Comment (enahum) [reply app/components/post/index.js]: I thought so too, but I wasn’t sure, let me change that

**Pull request merged**